girder/models/model_base.py

```diff
@@ -791,6 +791,15 @@ def _setAccess(self, doc, id, entity, level, save, flags=None, user=None, force=
             if '_id' not in doc:
                 doc = self.save(doc)
             else:
+                updateType = update.keys()[0]
+                # copy all other (potentially updated) fields to the update list
+                if updateType != '$set':
+                    update['$set'] = {prop: doc[prop] for prop in doc
+                                      if prop != 'access'}
```

Comment:
Slightly better would be:

    update['$set'] = {k: v for k, v in six.viewitems(doc) if k != 'access'}

(Saves a hash lookup for each key)